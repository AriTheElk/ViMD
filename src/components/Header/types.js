/* @flow */
export type HeaderProps = {
  onSave: () => {},
};
