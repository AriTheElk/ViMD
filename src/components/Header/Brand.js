import styled, { keyframes } from "styled-components";

import { fadeInLeft } from "../../styles/animations";

/**
 * Define styles for the brand name
 */
export default styled.h1`
  animation: ${fadeInLeft} 0.5s normal forwards;
  animation-delay: 0.5s;
  animation-timing-function: ease-out;
  color: #fff;
  font-size: 1.75em;
  font-style: italic;
  font-weight: 100;
  margin: 0 0 0 0px;
  opacity: 0;
  pointer-events: none;
  user-select: none;
  b {
    color: rgba(255, 255, 255, 0.4);
    font-weight: 100;
  }
`;
