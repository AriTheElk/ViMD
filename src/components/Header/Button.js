import styled from 'styled-components';

import Button from '../Button';

/**
 * Override the `Button` component with header specific styles
 * 
 * TODO: make the underline animate to the opposite side on mouseOut
 */
export default styled(Button)`
  border: none;
  margin: 0 10px;
  padding: 0 0 2px 0;
  position: relative;
  &::before {
    background: #fff;
    bottom: 0;
    color: transparent;
    content: ' ';
    height: 2px;
    position: absolute;
    transition: all .2s ease-in-out;
    width: 0;
  }
  &:hover {
    &::before {
      width: 100%;
    }
  }
`;
