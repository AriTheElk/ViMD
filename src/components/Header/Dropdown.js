import styled from "styled-components";
import ReactDropdown from "react-dropdown";
// import "react-dropdown/style.css";

const Dropdown = styled(ReactDropdown)`
  position: relative;

  .Dropdown-control {
    position: relative;
    overflow: hidden;
    border-radius: 2px;
    box-sizing: border-box;
    color: #fff;
    cursor: pointer;
    outline: none;
    padding: 8px 32px 8px 10px;
    transition: all 200ms ease;
  }

  .Dropdown-arrow {
    border-color: #fff transparent transparent;
    border-style: solid;
    border-width: 5px 5px 0;
    content: " ";
    display: block;
    height: 0;
    margin-top: -ceil(2.5);
    position: absolute;
    right: 10px;
    top: 22px;
    width: 0;
  }

  .is-open .Dropdown-arrow {
    border-color: transparent transparent #999;
    border-width: 0 5px 5px;
  }

  .Dropdown-menu {
    background-color: rgba(0, 0, 0, 0.8);
    box-shadow: 0 1px 0 rgba(0, 0, 0, 0.06);
    box-sizing: border-box;
    margin-top: -3px;
    max-height: 500px;
    overflow-y: auto;
    position: absolute;
    top: 100%;
    right: 0;
    width: 200px;
    z-index: 1000;
  }

  .Dropdown-option {
    box-sizing: border-box;
    color: #fff;
    cursor: pointer;
    display: block;
    padding: 8px 10px;

    &::last-child {
      border-bottom-right-radius: 2px;
      border-bottom-left-radius: 2px;
    }
  }
`;

export default Dropdown;
