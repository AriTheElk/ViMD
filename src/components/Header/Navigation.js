import styled, { keyframes } from 'styled-components';

import ButtonGroup from '../ButtonGroup';
import { fadeInTop } from '../../styles/animations';

/**
 * Add a fadeInTop animation to our navigation links
 */
export default styled(ButtonGroup)`
  animation: ${fadeInTop} .6s normal forwards;
  animation-timing-function: ease-out;
  animation-delay: 2s;
  opacity: 0;
`;
