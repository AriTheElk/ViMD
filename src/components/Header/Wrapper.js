import styled from "styled-components";
import Grid from "hedron";

/**
 * Define styles for the header wrapper
 */
export default styled(Grid.Bounds)`
  background: ${props => props.theme.header.background};
  line-height: 34px;
  height: ${props => props.theme.header.height};
  max-height: ${props => props.theme.header.height};
`;
