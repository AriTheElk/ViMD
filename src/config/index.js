export default {
    theme: 'tomorrow_night',
};
