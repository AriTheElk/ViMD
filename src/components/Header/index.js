import React from "react";
import Grid from "hedron";

import Brand from "./Brand";
import Button from "./Button";
import Navigation from "./Navigation";
import Link from "../Icon/Link";
import Wrapper from "./Wrapper";

/**
 * The `Header` component is responsible for rendering the top bar
 * with the brand name and menu navigation. The background color
 * and height can be set from within the `base.js` theme file.
 *
 * @param {Object} props
 * @returns React.node
 */
export default function Header(props) {
  const buttons = [
    <a href="https://github.com/garetmckinley/vimd" target="_blank">
      <Button key="github">GitHub</Button>
    </a>,
    <Button key="save" onClick={props.onSave}>
      Download
    </Button>,
  ];
  return (
    <Wrapper padding="10px" valign="center">
      <Grid.Box fluid>
        <Brand>
          Vi
          <b>MD</b>
        </Brand>
      </Grid.Box>
      <Grid.Box shiftRight fluid>
        <Navigation buttons={buttons} />
      </Grid.Box>
    </Wrapper>
  );
}
