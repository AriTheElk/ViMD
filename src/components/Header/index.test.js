// import chai from 'chai'
// import chaiEnzyme from 'chai-enzyme'
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import styled, { ThemeProvider } from 'styled-components';

import Header from '../Header';
import Brand from './Brand';
import config from '../../config';

const theme = require(`../../styles/themes/${config.theme}`).default;


it('renders header component', () => {
  const header = renderer.create(
    <ThemeProvider theme={theme}>
      <Header />
    </ThemeProvider>
  );
  let tree = header.toJSON();
  expect(tree).toMatchSnapshot();
});
