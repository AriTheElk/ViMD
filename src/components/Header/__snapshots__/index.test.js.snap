// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header component 1`] = `
<div
  className="sc-gqjmRU bIMQkO sc-bwzfXH gWBCma"
  direction="horizontal"
  wrap={false}
>
  <div
    className="sc-bxivhb ixCNyS"
    fill={false}
  >
    <h1
      className="sc-ifAKCX btOSSd"
    >
      Vi
      <b>
        MD
      </b>
    </h1>
  </div>
  <div
    className="sc-bxivhb bRsLbt"
    fill={false}
  >
    <ul
      className="sc-dnqmqq gJhIRP sc-gzVnrw chEyJs"
    >
      <li
        className="sc-htoDjs gYhcJG"
      >
        <button
          className="sc-bZQynM hNnjUf"
        >
          GitHub
        </button>
      </li>
      <li
        className="sc-htoDjs gYhcJG"
      >
        <button
          className="sc-bZQynM hNnjUf"
          onClick={undefined}
        >
          Save
        </button>
      </li>
    </ul>
  </div>
</div>
`;
